python/ql/test/library-tests/ApiGraphs/py3/test_annotations.py

```diff
@@ -0,0 +1,25 @@
+from types import AssignmentAnnotation, ParameterAnnotation
+
+def test_annotated_assignment():
+    local_x : AssignmentAnnotation = create_x() #$ MISSING: use=moduleImport("types").getMember("AssignmentAnnotation")
+    local_x #$ MISSING: use=moduleImport("types").getMember("AssignmentAnnotation").getAnnotatedInstance()
+
+global_x : AssignmentAnnotation #$ use=moduleImport("types").getMember("AssignmentAnnotation")
+global_x #$ MISSING: use=moduleImport("types").getMember("AssignmentAnnotation").getAnnotatedInstance()
```

Comment:
Why is this missing? Is it because there is no assignment on the line above, so that `global_x` is not in `getTarget` (which is presumably empty)?

Reply:
Hmm... This is a very salient question. If I quick-eval the `annotatedInstance` predicate, I get four results:
- `ControlFlowNode for ImportMember, ControlFlowNode for global_x`
- `ControlFlowNode for ImportMember, ControlFlowNode for parameter_y`
- `ControlFlowNode for Alias, ControlFlowNode for parameter_z`
- `ControlFlowNode for Alias, ControlFlowNode for global_z`

So, we _are_ picking up the instancing from `from ... import AssignmentAnnotation` to `global_x : AssignmentAnnotation`, but we're not picking up that the annotation is a use of that same identifier as in the `import` statement. What's curious, then, is that `global_x` _isn't_ seen as an instance of `AssignmentAnnotation`. For `global_z` it makes sense, since we don't understand the simple type aliasing that's taking place on line 13.

Looking at `getTarget` it _does_ exist for the type ascription of `global_x`, but it seems that we do not track the flow between the two occurrences of `global_x`. I'm trying to figure out why now.

Reply:
A thought occurred to me after writing that message. Could it be that we're observing that `global_x` gets overwritten here (because it's the target of an assignment), but then when we go to see what value was assigned we don't find it (because it's just a type ascription)? That would explain the weird behaviour.

Reply:
Yes that could be it. I wonder if `global_x` in `global_x : AssignmentAnnotation` should actually be considered a use rather than a def..